spec/command/apply_template_spec.rb

```diff
@@ -0,0 +1,173 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe Command::ApplyTemplate do
+  context "when any template does not exist" do
+    let!(:app) { dummy_test_app }
+
+    it "raises error", :fast do
+      result = run_cpl_command("apply-template", "gvc", "rails", "unexistent", "-a", app)
+
+      expect(result[:status]).to eq(1)
+      expect(result[:stderr]).to include("Missing templates")
+      expect(result[:stderr]).to include("- unexistent")
+    end
+  end
+
+  context "when all templates exist" do
+    let!(:app) { dummy_test_app }
+
+    after do
+      run_cpl_command!("delete", "-a", app, "--yes")
+    end
+
+    it "applies valid templates", :fast do
+      result = run_cpl_command("apply-template", "gvc", "rails", "-a", app)
+
+      expect(result[:status]).to eq(0)
+      expect(result[:stderr]).to include("Created items")
+      expect(result[:stderr]).to include("- [app] #{app}")
+      expect(result[:stderr]).to include("- [identity] #{app}-identity")
+      expect(result[:stderr]).to include("- [workload] rails")
+    end
+
+    it "fails to apply invalid templates", :fast do
+      result = run_cpl_command("apply-template", "invalid", "-a", app)
+
+      expect(result[:status]).to eq(1)
```

Comment:
Better exit codes, 64 and up, https://www.perplexity.ai/search/what-are-the-p2EQeuORQPyYDSLAjJ0RXQ

> While exit codes 1-2 are mentioned as having special meanings, most guides recommend using codes 64 and above for user-defined errors. This avoids conflicts with reserved codes.

> While there are no universal standards, many tools and scripts follow the convention of 0 for success and codes 64-78 for application-defined errors. Avoiding 1 helps consistency with other tools.
[4](https://superuser.com/questions/1662138/how-do-i-set-the-exit-code-of-a-one-line-command-to-1-if-the-result-if-blank)
[5](https://unix.stackexchange.com/questions/313875/how-can-i-set-a-return-code-from-the-command-line-without-exiting-my-ssh-client)